### ISSUE TITLE
Normalize doc comments

### ISSUE THREAD

**Issue by dtolnay** [rust-lang-nursery/rustfmt#2883]
I would like rustfmt to perform the following normalizations:

```diff
- #![doc = r" content"]
+ //! content

- #[doc = r" content"]
+ /// content
```

This would enable crates to use the proc-macro APIs to construct source code as tokens, run the code through rustfmt and write it to a file that is reasonably readable. The proc-macro API, like the macro_rules API, has no way to represent the distinction between doc attributes and doc comments because there really is no distinction when handing tokens over to the compiler to be directly compiled.

If necessary, this behavior could be behind a `normalize_doc_comments` or `normalize_doc_attributes` config setting.

**Comment by nrc** [comment]
Sounds like a good idea to me!

**Comment by lqd** [comment]
I believe this can be closed now that #3002 landed ?

**Comment by dtolnay** [comment]
Great! Thank you.
